doc/specs/vulkan/chapters/descriptorsets.txt

```diff
@@ -2373,6 +2373,12 @@ include::../api/protos/vkUpdateDescriptorSetWithTemplateKHR.txt[]
     of slink:VkDescriptorImageInfo, slink:VkDescriptorBufferInfo, or
     slink:VkBufferView used to write the descriptors.
 
+.Valid Usage
+****
+  * pname:pData must: be a pointer to a raw memory structure as defined by
+    pname:descriptorUpdateTemplate when it was created
```

Comment:
Wouldn't it make more sense to say "must: be a pointer to an array of instances of VkDescriptorImageInfo,
VkDescriptorBufferInfo, and VkBufferView structure."?

Ideally we should be more precise and have a number of statements indicating at which specific structures go where but... we should probably save that for a future PR :)

Reply:
I haven't study the extension in that much detail.
Still my version is sort of complete, but too terse.
Your version does reference the needed types, but does say pretty much nothing about the validity. What does "being a pointer to array of three types" even mean? How would you test it?

Reply:
> What does "being a pointer to array (of unspecified size) of three types" even mean? How would you test it?

The validation layers would be able to cast to an untyped Vulkan struct with "sType" in it, and thus could determine the size and type of the struct, then they have enough information to iterate to the next one and repeat until they run out. We could have a vague statement of "n structs according to the number of structs specified for the descriptor template" which is imprecise, but enough to go to the spec to figure out how many structs there should be in total.

So specifically that would be:

>  * pname:pData must: be a pointer to an array of [eq]#n# instances of VkDescriptorImageInfo, VkDescriptorBufferInfo, and VkBufferView structure, where [eq]#n# is the number of descriptors described by pname:descriptorUpdateTemplate.

It's not super precise, but it's enough to do some basic validation. We can iterate on more precise VU language after this PR is in...

Reply:
> The validation layers would be able to cast to an untyped Vulkan struct with "sType" in it, and thus could determine the size and type of the struct

`VkBufferView` is not a structure.

Reply:
Hm good point - that would change the wording a bit. But given that it's just VkBufferView that isn't a type, they'd see that it's not one of the sTypes, so they could check that the 64-bit value is a valid VkBufferView that's been created by the app. Not a big deal.

As for what "n" is, someone will have to look at the spec and figure it out from the definition of the descriptor templates. This would require extra work to actually define in the VU properly. Best we can do for now is make a link to the vkCreateDescriptorUpdateTemplateKHR function.

So updated wording:

>  * pname:pData must: be a pointer to an array of [eq]#n# instances of slink:VkDescriptorImageInfo, slink:VkDescriptorBufferInfo structures or slink:VkBufferView handles, where [eq]#n# is the number of descriptors described by pname:descriptorUpdateTemplate as defined by flink:vkCreateDescriptorUpdateTemplateKHR .

Reply:
@TobiasHector Well, we are slowly iterating it now anyway. Might as well figure out what the best wording is now. We have a saying in my country about a lazyman that died from overwork.

Reply:
Hmm, if we want to do that then (regrettably) I'd say we should split that work off - I'd rather get the misleading VU statements out sooner rather than waiting on us to iterate precise wording - plus it'll require additional feedback from the group. I currently have group mandate to sign this PR off on my own, but if we get to that level of detail I'll need to go back to the working group for other people to review.

So by all means, yea, let's do that, but in another PR. For this PR then we should remove the explicit VU for these parameters, if that's the route you want to go.

Reply:
@TobiasHector Nah, seems solid to me. I was just being sarcastic at the energy expended to figure out a quick placeholder, until your pretty much came with a version that looks more final than both initially proposed versions.

Reply:
Haha fair enough - sorry the subtlty was lost on me :D

Reply:
Actually, reading the extension there are also `offsets` and `strides` allowing for holes and whatnot.
I would drop the "n" and "array", which is not that meaningful concept in that context. So proposing another placeholder (until authors figure out what exactly they mean here):

> `pData` is a pointer to a memory which contains one or more valid `VkDescriptorImageInfo`, `VkDescriptorBufferInfo`, or `VkBufferView` in a layout as definded by `descriptorUpdateTemplate` when it was created with [vkCreateDescriptorUpdateTemplateKHR]().

Reply:
Perfect - let's go with that.